src/in_mem_graph_store.cpp

```diff
@@ -6,26 +6,238 @@
+    }
+}
+
+std::tuple<uint32_t, uint32_t, size_t> InMemGraphStore::load(const std::string &index_path_prefix,
+                                                             const size_t num_points)
+{
+    return load_impl(index_path_prefix, num_points);
+}
+int InMemGraphStore::store(const std::string &index_path_prefix, const size_t num_points,
+                           const size_t num_frozen_points, const uint32_t start)
+{
+    return save_graph(index_path_prefix, num_points, num_frozen_points, start);
+}
+const std::vector<location_t> &InMemGraphStore::get_neighbours(const location_t i) const
+{
+    return _graph.at(i);
+}
+
+void InMemGraphStore::add_neighbour(const location_t i, location_t neighbour_id)
+{
+    _graph[i].emplace_back(neighbour_id);
+    if (_graph[i].size() > _max_observed_degree)
+    {
+        _max_observed_degree = (uint32_t)_graph.size();
+    }
+}
+
+void InMemGraphStore::clear_neighbours(const location_t i)
+{
+    _graph[i].clear();
+};
+void InMemGraphStore::swap_neighbours(const location_t a, location_t b)
+{
+    _graph[a].swap(_graph[b]);
+};
+
+void InMemGraphStore::set_neighbours(const location_t i, std::vector<location_t> &neighbors)
+{
+    _graph[i].assign(neighbors.begin(), neighbors.end());
+}
+
+size_t InMemGraphStore::resize_graph(const size_t new_size)
+{
+    _graph.resize(new_size);
+    set_total_points(new_size);
+    return _graph.size();
+}
+
+void InMemGraphStore::clear_graph()
+{
+    _graph.clear();
+}
 
-InMemGraphStore::InMemGraphStore(const size_t max_pts) : AbstractGraphStore(max_pts)
+#ifdef EXEC_ENV_OLS
+std::tuple<uint32_t, uint32_t, size_t> InMemGraphStore::load_impl(AlignedFileReader &reader, size_t expected_num_points)
 {
+    size_t expected_file_size;
+    size_t file_frozen_pts;
+    uint32_t start;
+
+    auto max_points = get_max_points();
+    int header_size = 2 * sizeof(size_t) + 2 * sizeof(uint32_t);
+    std::unique_ptr<char[]> header = std::make_unique<char[]>(header_size);
+    read_array(reader, header.get(), header_size);
+
+    expected_file_size = *((size_t *)header.get());
+    _max_observed_degree = *((uint32_t *)(header.get() + sizeof(size_t)));
+    start = *((uint32_t *)(header.get() + sizeof(size_t) + sizeof(uint32_t)));
+    file_frozen_pts = *((size_t *)(header.get() + sizeof(size_t) + sizeof(uint32_t) + sizeof(uint32_t)));
+
+    diskann::cout << "From graph header, expected_file_size: " << expected_file_size
+                  << ", _max_observed_degree: " << _max_observed_degree << ", _start: " << start
+                  << ", file_frozen_pts: " << file_frozen_pts << std::endl;
+
+    diskann::cout << "Loading vamana graph from reader..." << std::flush;
+
+    // If user provides more points than max_points
+    // resize the _graph to the larger size.
+    if (get_total_points() < expected_num_points)
+    {
+        diskann::cout << "resizing graph to " << expected_num_points << std::endl;
+        _graph.resize(expected_num_points);
```

Comment:
shouldn't we call the resize function of this class instead of _graph?

Reply:
Yes. good catch. I have updated the code.